### PR TITLE
[BUILD] Share jars to reduce binary release tarball size

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

The 1.6.0 binary tarball is nearly 300MB, we can reduce the size by sharing jars between engine and server, as beeline does.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

```
-rw-r--r--   1 chengpan  staff   155M Sep  9 14:21 apache-kyuubi-1.7.0-SNAPSHOT-bin.tgz
```

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/dist [hunk 1]
```diff
@@ -314,7 +328,7 @@ if [[ "$MAKE_TGZ" == "true" ]]; then
   TARDIR="$KYUUBI_HOME/$TARDIR_NAME"
   rm -rf "$TARDIR"
   cp -R "$DISTDIR" "$TARDIR"
-  tar czf "$TARDIR_NAME.tgz" -C "$KYUUBI_HOME" "$TARDIR_NAME"
+  tar czf "$TARDIR.tgz" -C "$KYUUBI_HOME" "$TARDIR_NAME"
```
Comment (pan3793) [inline build/dist:331]: it changes the output tgz always under project home, not current home.

**Comment by hddong** [comment]
LGTM, add trino engine works well.

**Comment by pan3793** [comment]
Thanks, merging to master/1.6

**Comment by pan3793** [comment]
Thanks for verification

**Pull request closed without merging**